pkg/netutils/netutils_test.go

```diff
@@ -0,0 +1,1042 @@
+		if r.Dst.String() == "0.0.0.0/0" {
+			return true
+		}
+	}
+	return false
+}
+
+func test020ResultHasIPv6DefaultRoute(data []byte) bool {
+	resultRaw, err := types020.NewResult(data)
+	Expect(err).NotTo(HaveOccurred())
+
+	result, err := types020.GetResult(resultRaw)
+	Expect(err).NotTo(HaveOccurred())
+
+	for _, r := range result.IP6.Routes {
+		if r.Dst.String() == "::/0" {
+			return true
+		}
+	}
+	return false
+}
+
+func testResultHasIPv4DefaultRoute(data []byte) bool {
+	resultRaw, err := current.NewResult(data)
+	Expect(err).NotTo(HaveOccurred())
+
+	result, err := current.GetResult(resultRaw)
+	Expect(err).NotTo(HaveOccurred())
+
+	for _, r := range result.Routes {
+		if r.Dst.String() == "0.0.0.0/0" {
+			return true
+		}
+	}
+	return false
+}
+
+func testResultHasIPv6DefaultRoute(data []byte) bool {
+	resultRaw, err := current.NewResult(data)
+	Expect(err).NotTo(HaveOccurred())
+
+	result, err := current.GetResult(resultRaw)
+	Expect(err).NotTo(HaveOccurred())
+
+	for _, r := range result.Routes {
+		if r.Dst.String() == "::/0" {
+			return true
+		}
+	}
+	return false
+}
+
+var _ = Describe("netutil netlink function testing", func() {
+	const IFNAME string = "dummy0"
+	var IFMAC net.HardwareAddr = net.HardwareAddr([]byte{0x02, 0x66, 0x7d, 0xe3, 0x14, 0x1c})
+	var originalNS ns.NetNS
+	var targetNS ns.NetNS
+
+	BeforeEach(func() {
+		// Create a new NetNS so we don't modify the host
+		var err error
+		originalNS, err = testutils.NewNS()
+		Expect(err).NotTo(HaveOccurred())
+
+		targetNS, err = testutils.NewNS()
+		Expect(err).NotTo(HaveOccurred())
+
+		err = targetNS.Do(func(ns.NetNS) error {
+			defer GinkgoRecover()
+
+			err = netlink.LinkAdd(&netlink.Dummy{
+				LinkAttrs: netlink.LinkAttrs{
+					Name: IFNAME,
+					HardwareAddr: IFMAC,
+					Index: 10,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			_, err = netlink.LinkByName(IFNAME)
+			Expect(err).NotTo(HaveOccurred())
+			return nil
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		Expect(originalNS.Close()).To(Succeed())
+	})
+
+	Context("test DeleteDefaultGW", func() {
+		It("verify default gateway is removed", func() {
+ 			err := targetNS.Do(func(ns.NetNS) error {
+				defer GinkgoRecover()
+				link, err := netlink.LinkByName(IFNAME)
+				Expect(err).NotTo(HaveOccurred())
+				err = netlink.LinkSetUp(link)
+				Expect(err).NotTo(HaveOccurred())
+
+				// addr 10.0.0.2/24
+				err = testAddAddr(link, net.IPv4(10, 0, 0, 2), net.CIDRMask(24, 32))
+				Expect(err).NotTo(HaveOccurred())
+
+				// add default gateway into IFNAME
+				err = testAddRoute(link,
+					net.IPv4(0, 0, 0, 0), net.CIDRMask(0, 0),
+					net.IPv4(10, 0, 0, 1))
+				Expect(err).NotTo(HaveOccurred())
+
+				//"dst": "10.0.0.0/16"
+				err = testAddRoute(link,
+					net.IPv4(10, 0, 0, 0), net.CIDRMask(16, 32),
+					net.IPv4(10, 0, 0, 1))
+				Expect(err).NotTo(HaveOccurred())
+
+				return nil
+			})
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
Would it make sense to move all this test setup stage into a `BeforeEach` block ?

Reply:
Each test setup code is different, hence if we cannot use `BeforeEach`. Is there any way to change inside `BeforeEach` for each testing? (To me, I suppose we don't just use `BeforeEach` in such case...)